ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_repo-layout/settings.tsx

```diff
@@ -0,0 +1,275 @@
+                      onValueChange={field.onChange}
+                      defaultValue={field.value}
+                    >
+                      <FormControl>
+                        <SelectTrigger>
+                          <SelectValue placeholder='Select a type' />
+                        </SelectTrigger>
+                      </FormControl>
+                      <SelectContent>
+                        {Object.values($RepositoryType.enum).map((type) => (
+                          <SelectItem key={type} value={type}>
+                            {type}
+                          </SelectItem>
+                        ))}
+                      </SelectContent>
+                    </Select>
+                    <FormMessage />
+                  </FormItem>
+                )}
+              />
+            </CardContent>
+            <CardFooter>
+              <Button
+                type='button'
+                className='m-1'
+                variant='outline'
+                onClick={() =>
+                  navigate({
+                    to:
+                      '/organizations/' +
+                      params.orgId +
+                      '/products/' +
+                      params.productId,
+                  })
+                }
+                disabled={isPending}
+              >
+                Cancel
+              </Button>
+              <Button type='submit' disabled={isPending}>
+                {isPending ? (
+                  <>
+                    <span className='sr-only'>Editing repository...</span>
+                    <Loader2 size={16} className='mx-3 animate-spin' />
+                  </>
+                ) : (
+                  'Submit'
+                )}
+              </Button>
+            </CardFooter>
+          </form>
+        </Form>
+      </Card>
+      <Card>
+        <CardHeader>
+          <CardTitle>Danger Zone</CardTitle>
+        </CardHeader>
+        <CardContent>
+          <div className='flex justify-between'>
+            <div>Delete this repository</div>
+            <DeleteDialog
+              item={{
+                descriptor: 'repository',
+                name: repository.url,
+              }}
+              onDelete={handleDelete}
+              isPending={deleteIsPending}
+              trigger={<Button variant='destructive'>Delete repository</Button>}
+            />
+          </div>
+        </CardContent>
+      </Card>
+    </div>
+  );
+};
+
+export const Route = createFileRoute(
+  '/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_repo-layout/settings'
+)({
+  loader: async ({ context, params }) => {
+    await context.queryClient.ensureQueryData({
+      queryKey: [useRepositoriesServiceGetRepositoryByIdKey, params.repoId],
+      queryFn: () =>
+        RepositoriesService.getRepositoryById({
+          repositoryId: Number.parseInt(params.repoId),
+        }),
+    });
+  },
+  component: RepositorySettingsPage,
+});
```

Comment:
Same here: why not use the prefetch hook? Query key is also now different from the one above.

Reply:
True, seems to be a bug in the old edit page, which got carried over here. I'll add a fixing comment.